logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt

```diff
@@ -62,8 +64,10 @@ abstract class UserSessionScopeCommon(
             clientMapper
         )
 
+    private val clientRegistrationStorage: ClientRegistrationStorage = ClientRegistrationStorageImpl(userPreferencesSettings)
```

Comment:
if `clientRegistrationStorage` meant to be a singleton, can it be lazy initialized ?
```suggestion
    private val clientRegistrationStorage: ClientRegistrationStorage by lazy { ClientRegistrationStorageImpl(userPreferencesSettings) }
```

Reply:
I don't know what I had in my mind 😄. An oversight from my part, absolutely. Nice catch!